### DISCUSSION TITLE
Fauxton: Mango - Help with retrieving previous revisions of records (Remove newer revisions based on criteria)

### DISCUSSION THREAD

**Discussion by Zero2XIII** [Budibase/budibase#10157] [Help] [answered]
Hi All,

So I ran into a glitch/undisclosed feature of budibase (See [here](https://github.com/Budibase/budibase/discussions/5883) and [here](https://github.com/Budibase/budibase/discussions/5442)) where the development data and the production data was not "in sync" and when I published new changes to the GUI a lot of the data in the couchDB instance was altered (back to the version in the developement db).

There is very limitied information on the Fauxton/Mango process on quering the DB and working with data and finding tools that support couchDB is proving to be a challenge (Especially trying to connect in a docker environment.)

Anyway, the short version of what I need is to "delete" newer entries back to previous versions where there were more data available. See the following:

Query
`{
   "selector": {
      "Vin Number": {
         "$eq": "WV1ZZZ7HZPH005454"
      }
   }
}`

Result
`{
 "_id": "ro_ta_2b5f0641d0da4b188296ab6b36dcc4ad_0a0bd13df3c747fb8fdf2af879cf8362",
 "_rev": "4-6f2311e7deb22996d2b408936f40a691",
 "Client": "VW Witbank",
 "Quote Number": "",
 "Conversion Type": null,
 "Vehicle Make": "Volkswagen",
 "Vehicle Model": "Transporter",
 "Expected Date of Arrival": "2022-09-28T00:00:00.000",
 "Expected Delivery Date": null,
 "Logistic Status": "Delivered",
 "Vin Number": "WV1ZZZ7HZPH005454",
 "tableId": "ta_2b5f0641d0da4b188296ab6b36dcc4ad",
 "Auto ID": 156,
 "Created At": "2022-10-05T07:24:36.793Z",
 "Updated At": "2022-12-12T11:16:26.662Z",
 "type": "row",
 "VMS": "H005454",
 "Actual Arrival Date": "October 05, 2022",
 "Production Start Date": null,
 "Production Expected End Date": null,
 "Production Actual End Date": null,
 "Delivery Date": "December 12, 2022",
 "Invoice Number": "NOTHING DONE ON VEHICLE",
 "Delivery Note": [
  {
   "size": 656871,
   "name": "VW Witbank - H005454.pdf",
   "url": "/prod-budi-app-assets/app_9c9546b4bc16451bb2e5266e51d27e4f/attachments/22a909b6-16ec-4e49-954a-0748b347e105.pdf",
   "extension": "pdf",
   "key": "app_9c9546b4bc16451bb2e5266e51d27e4f/attachments/22a909b6-16ec-4e49-954a-0748b347e105.pdf"
  },
  {
   "size": 656871,
   "name": "VW Witbank - H005454.pdf",
   "url": "/prod-budi-app-assets/app_9c9546b4bc16451bb2e5266e51d27e4f/attachments/2d16c0bb-5869-4ebd-877b-c57434c77b86.pdf",
   "extension": "pdf",
   "key": "app_9c9546b4bc16451bb2e5266e51d27e4f/attachments/2d16c0bb-5869-4ebd-877b-c57434c77b86.pdf"
  }
 ],
 "Job Number": "",
 "createdAt": "2022-12-12T11:16:27.915Z",
 "updatedAt": "2022-12-12T11:16:27.915Z"
}

{
 "_id": "ro_ta_2b5f0641d0da4b188296ab6b36dcc4ad_bf284ecefd2a4ce0b6b52bc0634ef6f0",
 "_rev": "6-4216e2ed11cef42343d87ad76863f83c",
 "Client": "VW Witbank",
 "Quote Number": "",
 "Conversion Type": null,
 "Vehicle Make": "Volkswagen",
 "Vehicle Model": "Transporter",
 "Expected Date of Arrival": "2023-03-02T00:00:00.000",
 "Expected Delivery Date": "2023-03-02T00:00:00.000",
 "Logistic Status": "Delivered",
 "Vin Number": "WV1ZZZ7HZPH005454",
 "tableId": "ta_2b5f0641d0da4b188296ab6b36dcc4ad",
 "Auto ID": 550,
 "Created At": "2023-03-02T11:34:24.429Z",
 "Updated At": "2023-03-29T19:34:02.887Z",
 "type": "row",
 "VMS": "H005454",
 "Invoice Number": "",
 "Actual Arrival Date": "2000-03-01T22:00:00.000Z",
 "Production Start Date": null,
 "Production Expected End Date": null,
 "Production Actual End Date": null,
 "Delivery Date": null,
 "eNatis Number": "",
 "Delivery Date Text": "",
 "Production Status": null,
 "Delivery Note": [],
 "QC Status": null,
 "Vehicle At Station": "",
 "Job Card Issued Date": null,
 "Job Number": "",
 "createdAt": "2023-03-29T19:34:04.354Z",
 "updatedAt": "2023-03-29T19:34:04.354Z"
}`

I want the "live" data to be that of revision "4-6f2311e7deb22996d2b408936f40a691" as above.

I can bank on the following statements being true:

- Logistic Status = Delivered
- Invoice Number is not blank
- Delivery Note is not blank

It would be fantastic if it can be generalized to a "Which ever revision has the most data" statement.
If you can point me to decent manuals on how to achieve this it will also be helpful.

On a side note - any advice on how to "sync" the development environment data first before making this blunder again will be amazing.

I beg and thank in advance.

**Comment by Zero2XIII** [comment]
Sorry missed to add all the technicals:
Budibase Version: 2.0.31
CouchDB Version: 3.2.1
Community Edition - No License
Docker Compose Deployment, Windows Environment

**Comment by melohagan** [comment]
Hey @Zero2XIII 

It sounds like maybe you need to resolve conflicts in your data? There's a guide on doing that here: https://github.com/Budibase/budibase/discussions/7777#discussioncomment-3732390

**Reply by Zero2XIII** [reply]
Hi @melohagan 

There are no conflicts, see screenshots. Any other suggestions?

![image](https://user-images.githubusercontent.com/114136925/228943274-c10511ae-312a-410a-8c8a-307f0a8480b6.png)


![image](https://user-images.githubusercontent.com/114136925/228943201-50f58f0e-3d34-4fa2-ae06-b7d404d55813.png)

**Comment by Zero2XIII** [comment, marked as answer]
Hi Guys,

So I managed to resolve this with a very manual process.

I created a view in Fauxton with the following:
`function(doc) {
  if (doc['_conflicts'] && doc['Vin Number']) {
    emit(doc['Vin Number'], doc._id);
  }
}`

This gave me a list of records and when I open them up, you can see the conflict and decide which to keep. Key being opening up the entry and not relying on the visual que as that tends to come and go. They GUI for Fauxton is absurdly buggy, but I managed to resolve over a 100 conflicts by hand in doing so, which was a success...
